### PR TITLE
Autofill LastLedgerSequence for multisigned transactions

### PR DESCRIPTION
**Pull request by wltsmrz**
+ `Signers` has been moved to tx_json for submit/submit_multisigned
+ Automatically set LastLedgerSequence, honor existing values
+ By default, LastLedgerSequence offset is unrealistically low. Users should call either `tx.setLastLedgerSequence()` or `tx.setLastLedgerSequenceOffset()` before `tx.getMultiSigningJson()` / multisigning

### CODE REVIEW

File: src/core/transaction.js [hunk 1]
```diff
@@ -1512,30 +1524,44 @@ Transaction.prototype.setSigners = function(signers) {
...
   assert(this.tx_json.Fee, 'Fee must be set before multi-signing');
 
-  const signingTx = Transaction.from_json(this.tx_json);
+  if (_.isUndefined(this.tx_json.LastLedgerSequence)) {
+    // Auto-fill LastLedgerSequence
+    this.setLastLedgerSequence();
+  }
+
+  const cleanedJson = _.omit(this.tx_json, [
+    'SigningPubKey',
```
Comment (sublimator) [inline src/core/transaction.js]: The empty SigningPubKey needs to be signed

File: src/core/transaction.js [hunk 2]
```diff
@@ -1512,30 +1524,44 @@ Transaction.prototype.setSigners = function(signers) {
...
+    'Signers',
+    'TxnSignature'
+  ]);
+
+  const signingTx = Transaction.from_json(cleanedJson);
   signingTx.remote = this.remote;
   signingTx.setSigningPubKey('');
```
Comment (sublimator) [inline src/core/transaction.js]: Oh, it's set here huh

File: src/core/transaction.js [hunk 3]
```diff
@@ -582,19 +583,30 @@ Transaction.prototype.clientID = function(id) {
...
-  this._setUInt32('LastLedgerSequence', sequence);
+Transaction.prototype.setLastLedgerSequence = function(sequence) {
+  if (!_.isUndefined(sequence)) {
+    this._setUInt32('LastLedgerSequence', sequence);
+  } else {
+    // Autofill LastLedgerSequence
+    assert(this.remote, 'Unable to set LastLedgerSequence, missing Remote');
+
+    this._setUInt32('LastLedgerSequence',
+                    this.remote.getLedgerSequence() + 1
```
Comment (sublimator) [inline src/core/transaction.js]: getLedgerSequence is .... ? the last validated ledger seen ?

**Comment by scottschurr** [comment]
@wltsmrz It looks like `submit_multisign` is working for you with the `Signers` moved inside the `tx_json`.  Is that a go for me to submit a pull request for that change in RippleD?

**Comment by wltsmrz** [comment]
Yep, looks good

File: src/core/transaction.js (hunk 3 above)
Comment (wltsmrz) [inline src/core/transaction.js]: Yeah

File: src/core/transaction.js (hunk 2 above)
Comment (wltsmrz) [inline src/core/transaction.js]: Indeed

**Comment by wltsmrz** [comment]
Adding a note that this will probably break if the following pull request is mergedhttps://github.com/ripple/ripple-lib/pull/546

**Pull request merged**